avocado/core/safeloader/docstring.py

```diff
@@ -1,6 +1,8 @@
 import json
 import re
 
+from avocado.plugins.dependency import Dependency
```

Comment:
`avocado.core` should not be importing from `avocado.plugins`, it should be the opposite.  The easy fix is to move `class Dependency` to, say, `avocado.core.nrunner.dependency`.

Reply:
Thank you for catching this, I was wondering why it was not caught by static-checks, but then I found out that we don't have such test. Therefore, I have created #5909

Reply:
Actually, I have looked it this more deeply, and I ponder if the safeloader should be in avocado.core. I understand that it is from historical reasons when avocado-instrumented test were part of legacy runner. But now the safeloader is used only by resolvers, and it is closely connected to them. Therefore, IMO it should be moved to plugins, but maybe there is something which I don't see. What do you think @clebergnu?

Reply:
I think the safeloader provides a generic core functionality, that is, the ability to parse Python code and find "stuff" without running them.  For now, I'd keep it where it is.